src/routes/transactions/transactions-history.imitation-transactions.controller.spec.ts

```diff
@@ -100,1393 +105,2586 @@ describe('Transactions History Controller (Unit) - Imitation Transactions', () =
+        const results = [
+          valueIntolerantIncomingTransaction,
+          valueIntolerantOutgoingTransaction,
+          notImitatedMultisigTransaction,
+          notImitatedMultisigTransaction,
+          multisigTransaction,
+        ];
+        networkService.get.mockImplementation(({ url }) => {
+          if (url === `${safeConfigUrl}/api/v1/chains/${chain.chainId}`) {
+            return Promise.resolve({ data: chain, status: 200 });
+          }
+          if (url === getAllTransactionsUrl) {
+            return Promise.resolve({
+              data: pageBuilder().with('results', results).build(),
+              status: 200,
+            });
+          }
+          if (url === getSafeUrl) {
+            return Promise.resolve({ data: safe, status: 200 });
+          }
+          if (url === getTokenAddressUrl) {
+            return Promise.resolve({
+              data: multisigToken,
+              status: 200,
+            });
+          }
+          if (url === getNotImitatedTokenAddressUrl) {
+            return Promise.resolve({
+              data: notImitatedMultisigToken,
+              status: 200,
+            });
+          }
+          if (url === getImitationTokenAddressUrl) {
+            return Promise.resolve({
+              data: imitationToken,
+              status: 200,
+            });
+          }
+          return Promise.reject(new Error(`Could not match ${url}`));
         });
-      }
-      return Promise.reject(new Error(`Could not match ${url}`));
-    });
 
-    await request(app.getHttpServer())
-      .get(
-        `/v1/chains/${chain.chainId}/safes/${safe.address}/transactions/history?trusted=false&imitation=false`,
-      )
-      .expect(200)
-      .then(({ body }) => {
-        expect(body.results).toStrictEqual([
-          {
-            timestamp: 1710927778000,
-            type: 'DATE_LABEL',
-          },
-          {
-            conflictType: 'None',
-            transaction: {
-              executionInfo: null,
-              // @ts-expect-error - Type does not contain transfers
-              id: `transfer_${safe.address}_${results[0].transfers[0].transferId}`,
-              safeAppInfo: null,
-              timestamp: 1710927778000,
-              txInfo: {
-                direction: 'INCOMING',
-                humanDescription: null,
-                recipient: {
-                  logoUri: null,
-                  name: null,
-                  value: safe.address,
-                },
-                richDecodedInfo: null,
-                sender: {
-                  logoUri: null,
-                  name: null,
-                  value: imitationAddress,
+        await request(app.getHttpServer())
+          .get(
+            `/v1/chains/${chain.chainId}/safes/${safe.address}/transactions/history?trusted=false`,
+          )
+          .expect(200)
+          .then(({ body }) => {
+            expect(body.results).toStrictEqual([
```

Comment:
Since these expected values and test inputs are making the test file large and difficult to review, do you think it would be a good idea to move them to a separate file? For example:
`transactions-history.imitation-transactions.controller.spec.ts`
`transactions-history.imitation-transactions.mock.ts`

Reply:
What do you think about eventually splitting the suites per attack vector? Preferably this would be done after the follow up PR though.

Reply:
Sure, It makes sense to do it in a separate PR as a clean up